From c4e2f6203f92a1f903573fd700b34c162d60d30e Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 20 Jan 2022 15:19:13 +0200
Subject: [PATCH] imx8mp: compulab: Add separate device tree for each video
 output

CompuLab imx8mp SOMs have 3 lcdif interfaces that all enabled by
default. In order to use one specific (mipi,lvds or hdmi) iface only the
rest of interfaces must be disabled. Provided a separate file for each
iface type.

Iface enumeration:
lcdif1 -- mipi
lcdif2 -- lvds
lcdif3 -- hdmi

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile         |  6 +++
 .../boot/dts/compulab/som-imx8m-plus-hdmi.dts | 37 +++++++++++++++++++
 .../boot/dts/compulab/som-imx8m-plus-lvds.dts | 37 +++++++++++++++++++
 .../boot/dts/compulab/som-imx8m-plus-mipi.dts | 37 +++++++++++++++++++
 .../boot/dts/compulab/som-imx8m-plus.dts      |  3 ++
 .../boot/dts/compulab/ucm-imx8m-plus-hdmi.dts | 37 +++++++++++++++++++
 .../boot/dts/compulab/ucm-imx8m-plus-lvds.dts | 37 +++++++++++++++++++
 .../boot/dts/compulab/ucm-imx8m-plus-mipi.dts | 37 +++++++++++++++++++
 8 files changed, 231 insertions(+)
 create mode 100644 arch/arm64/boot/dts/compulab/som-imx8m-plus-hdmi.dts
 create mode 100644 arch/arm64/boot/dts/compulab/som-imx8m-plus-lvds.dts
 create mode 100644 arch/arm64/boot/dts/compulab/som-imx8m-plus-mipi.dts
 create mode 100644 arch/arm64/boot/dts/compulab/ucm-imx8m-plus-hdmi.dts
 create mode 100644 arch/arm64/boot/dts/compulab/ucm-imx8m-plus-lvds.dts
 create mode 100644 arch/arm64/boot/dts/compulab/ucm-imx8m-plus-mipi.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index 4f1c53dd794d..64d25671aa50 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -3,11 +3,17 @@ dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-thermal.dtb
 dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus_mipi-csi1.dtb
 dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus_mipi-csi2.dtb
 dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus_mipi-csi1-imx219.dtb
+dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-mipi.dtb
+dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-lvds.dtb
+dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-hdmi.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus.dtb
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-thermal.dtb
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus_mipi-csi1.dtb
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus_mipi-csi2.dtb
+dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-mipi.dtb
+dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-lvds.dtb
+dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-hdmi.dtb
 
 always-y       := $(dtb-y)
 subdir-y       := $(dts-dirs)
diff --git a/arch/arm64/boot/dts/compulab/som-imx8m-plus-hdmi.dts b/arch/arm64/boot/dts/compulab/som-imx8m-plus-hdmi.dts
new file mode 100644
index 000000000000..47f564d5d3e1
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/som-imx8m-plus-hdmi.dts
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "som-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab SOM-iMX8M-Plus (hdmi)";
+	compatible = "compulab,som-imx8m-plus", "fsl,imx8mp";
+};
+
+/* mipi dsi */
+&lcdif1 {
+	status = "disabled";
+};
+
+/* lvds */
+&lcdif2 {
+	status = "disabled";
+};
+
+/* hdmi */
+&lcdif3 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/compulab/som-imx8m-plus-lvds.dts b/arch/arm64/boot/dts/compulab/som-imx8m-plus-lvds.dts
new file mode 100644
index 000000000000..20a66622ce15
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/som-imx8m-plus-lvds.dts
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "som-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab SOM-iMX8M-Plus (lvds)";
+	compatible = "compulab,som-imx8m-plus", "fsl,imx8mp";
+};
+
+/* mipi dsi */
+&lcdif1 {
+	status = "disabled";
+};
+
+/* lvds */
+&lcdif2 {
+	status = "okay";
+};
+
+/* hdmi */
+&lcdif3 {
+	status = "disabled";
+};
diff --git a/arch/arm64/boot/dts/compulab/som-imx8m-plus-mipi.dts b/arch/arm64/boot/dts/compulab/som-imx8m-plus-mipi.dts
new file mode 100644
index 000000000000..f1e0242c797c
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/som-imx8m-plus-mipi.dts
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "som-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab SOM-iMX8M-Plus (mipi)";
+	compatible = "compulab,som-imx8m-plus", "fsl,imx8mp";
+};
+
+/* mipi dsi */
+&lcdif1 {
+	status = "okay";
+};
+
+/* lvds */
+&lcdif2 {
+	status = "disabled";
+};
+
+/* hdmi */
+&lcdif3 {
+	status = "disabled";
+};
diff --git a/arch/arm64/boot/dts/compulab/som-imx8m-plus.dts b/arch/arm64/boot/dts/compulab/som-imx8m-plus.dts
index 23f06119ce65..e42f75d0883e 100644
--- a/arch/arm64/boot/dts/compulab/som-imx8m-plus.dts
+++ b/arch/arm64/boot/dts/compulab/som-imx8m-plus.dts
@@ -613,14 +613,17 @@
 	status = "okay";
 };
 
+/* mipi dsi */
 &lcdif1 {
 	status = "okay";
 };
 
+/* lvds */
 &lcdif2 {
 	status = "okay";
 };
 
+/* hdmi */
 &lcdif3 {
 	status = "okay";
 
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-hdmi.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-hdmi.dts
new file mode 100644
index 000000000000..35aa894fa5f9
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-hdmi.dts
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "ucm-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab UCM-iMX8M-Plus (hdmi)";
+	compatible = "compulab,ucm-imx8m-plus", "fsl,imx8mp";
+};
+
+/* mipi dsi */
+&lcdif1 {
+	status = "disabled";
+};
+
+/* lvds */
+&lcdif2 {
+	status = "disabled";
+};
+
+/* hdmi */
+&lcdif3 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-lvds.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-lvds.dts
new file mode 100644
index 000000000000..e77eecef59be
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-lvds.dts
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "ucm-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab UCM-iMX8M-Plus (lvds)";
+	compatible = "compulab,ucm-imx8m-plus", "fsl,imx8mp";
+};
+
+/* mipi dsi */
+&lcdif1 {
+	status = "disabled";
+};
+
+/* lvds */
+&lcdif2 {
+	status = "okay";
+};
+
+/* hdmi */
+&lcdif3 {
+	status = "disabled";
+};
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-mipi.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-mipi.dts
new file mode 100644
index 000000000000..78c0ff9e29e9
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-mipi.dts
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2021 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "ucm-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab UCM-iMX8M-Plus (mipi)";
+	compatible = "compulab,ucm-imx8m-plus", "fsl,imx8mp";
+};
+
+/* mipi dsi */
+&lcdif1 {
+	status = "okay";
+};
+
+/* lvds */
+&lcdif2 {
+	status = "disabled";
+};
+
+/* hdmi */
+&lcdif3 {
+	status = "disabled";
+};
-- 
2.17.1

